sklearn/tests/test_cross_validation.py

```diff
@@ -95,7 +95,7 @@ def fit(self, X, Y=None, sample_weight=None, class_prior=None,
     def predict(self, T):
         if self.allow_nd:
             T = T.reshape(len(T), -1)
-        return T.shape[0]
+        return T[:, 0]
```

Comment:
I think it was never used. Otherwise the tests should have failed. I might have used the weird semantics at some point to, not sure who introduced them, I might be guilty.

I agree, the previous implementation seems very suspicious, so I didn't feel to bad about changing it.